.circleci/config.yml

```diff
@@ -245,8 +245,28 @@ jobs:
             python3 -m pip install -e .
             python3 -m pip install -r requirements.dev.txt
             python3 -m pip install gitpython==3.1.0
+
+            # GET THE BRANCH THAT THE CURRENT PR IS BEING MERGED INTO.
+            # THIS IS ONLY RELEVANT FOR CI RUNS, WHICH IS WHY IT'S NOT PART OF THE PYTHON CODE.
+            if [ -n "${CIRCLE_PULL_REQUEST}" ]; then
+              PR_NUMBER=$(basename ${CIRCLE_PULL_REQUEST})
+              echo "Pull Request Number: ${PR_NUMBER}"
+      
+              # Use GitHub API to get the target branch name of the PR using GITHUB_CLI_TOKEN
+              BRANCH_MERGED_INTO=$(curl -s -H "Authorization: token ${GITHUB_CLI_TOKEN}" \
+                "https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${PR_NUMBER}" \
+                | jq -r '.base.ref')
+              echo "Branch merged into: ${BRANCH_MERGED_INTO}"
+            
+              # This is required since it's possible this branch would not be available locally
+              git fetch origin ${BRANCH_MERGED_INTO}:${BRANCH_MERGED_INTO}
+            else
+              echo "This job does not appear to be running as part of a pull request."
+            fi
+
+            # TESTS
             coverage run --source=super_gradients -m unittest tests/breaking_change_tests/unit_test.py
-            coverage run --source=super_gradients -m unittest tests/breaking_change_tests/test_detect_breaking_change.py
+            coverage run --source=super_gradients -m unittest tests/breaking_change_tests/test_detect_breaking_change.py --branch_merged_into=${BRANCH_MERGED_INTO}
```

Comment:
I did this as part of the CI because it's ONLY relevant in the CI.

Reply:
Should we set BRANCH_MERGED_INTO to default `master` if the CIRCLE_PULL_REQUEST is not set?
The call `            coverage run --source=super_gradients -m unittest tests/breaking_change_tests/test_detect_breaking_change.py --branch_merged_into=${BRANCH_MERGED_INTO}` seems to be unconditionally attempt to read from this variable so I wonder what would happen if it's not available.

Reply:
If it's not available, the default is `master` in the python side.
But it's true that it would better be explicit. I changed it here [e93be17](https://github.com/Deci-AI/super-gradients/pull/1720/commits/e93be17641223a74bd3a73cf5a3f6c1910d6c026)